### PR TITLE
magento/magento2#31206: Selected shipping method does not contain val…

### LINKED ISSUE #31206: [GraphQl] Selected shipping method does not contain value including tax.
### Preconditions (*)

1. Magento 2.4.1 (also present in at least 2.3.5p2)

### Steps to reproduce (*)

1. Enable the shipping method flatrate and set its price to 10
2. Configure a default tax class for shipping that's not 0%. My example is 21%
2. Set tax/calculation/shipping_includes_tax to "Including Tax"
3. Set tax/cart_display/shipping to "Including Tax"
4. Set tax/display/shipping to "Including Tax"
4. Create a Cart through GraphQl
5. Add a product to your cart through GraphQl
6. Set a Shipping Address on your cart through GraphQl
7. Set flatrate/flatrate as your shipping method through GraphQl using the following query.
````
mutation setShippingMethodsOnCart(
    $cartId: String!, 
) {
    setShippingMethodsOnCart(
        input: {
            cart_id: $cartId, 
            shipping_methods:[{
		carrier_code:"flatrate",
		method_code:"flatrate"
	    }]
        }
    ) {
        cart {
            shipping_addresses {
                selected_shipping_method {
                    amount {
                        currency
                        value
                    }
                }
            }
        }
    }
}
````

### Expected result (*)

````
{
    "data": {
        "setShippingMethodsOnCart": {
            "cart": {
                "shipping_addresses": [
                    {
                        "selected_shipping_method": {
                            "amount": {
                                "currency": "EUR",
                                "value": 10.00
                            }
                        }
                    }
                ]
            }
        }
    }
}
````

### Actual result (*)

````
{
    "data": {
        "setShippingMethodsOnCart": {
            "cart": {
                "shipping_addresses": [
                    {
                        "selected_shipping_method": {
                            "amount": {
                                "currency": "EUR",
                                "value": 8.26
                            }
                        }
                    }
                ]
            }
        }
    }
}
````
There is currently no way to get the value of the selected shipping method including tax. This is a value you want for all countries where its normal to display prices including tax (most if not all EU countries) . Currently the only way to get the value (without code changes) is to match the selected shipping method with available shipping methods and get the price incl tax from there. 

I'm not certain if the way forward is giving both values to the endpoint or based on configuration change the response of selected_shipping_method { amount { value }}.

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [x] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by sasha19957099**
### Description (*)
Added values to GraphQL schema price_excl_tax price_incl_tax for Selected shipping method 

### Fixed Issues (if relevant)
1. Fixes https://github.com/magento/magento2/issues/31206

### Manual testing scenarios (*)
1.Enable the shipping method flatrate and set its price to 10
2. Configure a default tax class for shipping that's not 0%. My example is 21%
3. Set tax/calculation/shipping_includes_tax to "Including Tax"
4. Set tax/cart_display/shipping to "Including Tax"
5. Set tax/display/shipping to "Including Tax"
6. Create a Cart through GraphQl
7.Add a product to your cart through GraphQl
8.Set a Shipping Address on your cart through GraphQl
9. Set flatrate/flatrate as your shipping method through GraphQl using the following query.

mutation setShippingMethodsOnCart($cartId: String!) {
  setShippingMethodsOnCart(input: {cart_id: $cartId, shipping_methods: [{carrier_code: "flatrate", method_code: "flatrate"}]}) {
    cart {
      shipping_addresses {
        selected_shipping_method {
          amount {
            currency
            value
          }
          price_excl_tax {
            currency
            value
          }
          price_incl_tax {
            currency
            value
          }
        }
        available_shipping_methods{
          amount {
            currency
            value
          }
          price_excl_tax {
            currency
            value
          }
          price_incl_tax {
            currency
            value
          }
        }
      }
    }
  }
}

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by sasha19957099** [comment]
@magento run all tests

**Comment by victorchiriac89** [comment]
Any news over this, when will it be included?

**Review by cpartica — APPROVED** [review]

File: app/code/Magento/QuoteGraphQl/etc/schema.graphqls [hunk 1]
```diff
@@ -263,6 +263,8 @@ type SelectedShippingMethod {
     method_title: String!
     amount: Money!
```
Comment (cpartica) [inline app/code/Magento/QuoteGraphQl/etc/schema.graphqls:264]: here amount should be deprecated

Comment (cpartica) [inline app/code/Magento/QuoteGraphQl/etc/schema.graphqls:264]: amount should either be removed or renamed to display_price

File: app/code/Magento/QuoteGraphQl/etc/schema.graphqls [hunk 2]
```diff
@@ -263,6 +263,8 @@ type SelectedShippingMethod {
...
     base_amount: Money @deprecated(reason: "The field should not be used on the storefront")
+    price_excl_tax: Money!
```
Comment (cpartica) [inline app/code/Magento/QuoteGraphQl/etc/schema.graphqls]: **First option:**
'price' word should be change to 'amount' resulting in
amount_excl_tax: Money!
amount__incl_tax: Money!

**Second option:**
amount and prices can't be used interchangeably
amount should either be removed or renamed to display_price or removed

**Review by cpartica — APPROVED** [review]

**Comment by mzeis** [comment]
@cpartica Will this be fixed in 2.4.3? Thank you!

**Comment by sasha19957099** [comment]
@magento run Functional Tests B2B

**Review by engcom-Alfa — CHANGES_REQUESTED** [review]
Hi @sasha19957099 Thanks for your contribution and collaboration!
As I have checked the issue, it is reproducible in 2.4-develop.
There are Merge conflicts in this PR request, in order to process it further the conflicts should be resolved. Kindly resolve it and re update please!

**Comment by engcom-Lima** [comment]
@magento run all tests

**Comment by engcom-Lima** [comment]
@magento run Functional Tests B2B,Functional Tests CE,Functional Tests EE,Static Tests

**Comment by engcom-Lima** [comment]
@magento rull all tests

**Comment by engcom-Lima** [comment]
@magento run all tests

**Comment by engcom-Lima** [comment]
@magento run Functional Tests B2B,Functional Tests CE,Functional Tests EE,Static Tests

**Comment by engcom-Lima** [comment]
@magento run all tests

**Comment by engcom-Lima** [comment]
@magento run all tests

**Comment by engcom-Lima** [comment]
@magento run Functional Tests B2B,Functional Tests CE,Integration Tests

**Comment by engcom-Lima** [comment]
@magento run Functional Tests B2B,Integration Tests

**Comment by engcom-Lima** [comment]
@magento run Integration Tests

**Comment by engcom-Lima** [comment]
@magento run Integration Tests

**Comment by engcom-Lima** [comment]
@magento run Integration Tests

**Comment by engcom-Lima** [comment]
Hi @cpartica, after resolving the conflicts and running the latest build there were some static failures which were addressed.
Can you please review the changes again?
Thanks

**Review by ishakhsuvarov — APPROVED** [review]

**Comment by engcom-Alfa** [comment]
@magento run all tests

**Comment by engcom-Alfa** [comment]
@magento run Functional Tests B2B, Functional Tests EE, Integration Tests

**Comment by engcom-Alfa** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by engcom-Alfa** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by engcom-Charlie** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by engcom-Charlie** [comment]
As the build is green now and as per this [comment](https://github.com/magento/magento2/pull/31322#issuecomment-1110942520) PR is QA passed so moving it to Merge in Progress.

Thank you!

**Pull request merged**